st2api/st2api/controllers/v1/keyvalue.py

```diff
@@ -42,25 +43,36 @@ class KeyValuePairController(ResourceController):
     model = KeyValuePairAPI
     access = KeyValuePair
     supported_filters = {
-        'prefix': 'name__startswith'
+        'prefix': 'name__startswith',
+        'scope': 'scope'
     }
 
     def __init__(self):
         super(KeyValuePairController, self).__init__()
         self._coordinator = coordination.get_coordinator()
         self.get_one_db_method = self._get_by_name
 
-    @jsexpose(arg_types=[str, bool])
-    def get_one(self, name, decrypt=False):
+    @jsexpose(arg_types=[str, str, bool])
+    def get_one(self, name, scope=SYSTEM_KV_PREFIX, decrypt=False):
         """
             List key by name.
 
             Handle:
                 GET /keys/key1
         """
         from_model_kwargs = {'mask_secrets': not decrypt}
-        kvp_db = super(KeyValuePairController, self)._get_one_by_name_or_id(name_or_id=name,
-            from_model_kwargs=from_model_kwargs)
+        if scope:
+            kwargs = {'name': name, 'scope': scope}
+            kvp_db = super(KeyValuePairController, self)._get_all(
+                from_model_kwargs=from_model_kwargs,
+                **kwargs
+            )
+            kvp_db = kvp_db[0] if kvp_db else None
```

Comment:
>why/how would get_one by name+scope return multiple values?

It is not returning more than one value. It simply returns an array with one item which is both counter intuitive and blows up CLI.